app/coffee/Features/Project/ProjectDetailsHandler.coffee

```diff
@@ -39,16 +38,29 @@ module.exports =
 			callback(err)
 
 	renameProject: (project_id, newName, callback = ->)->
-		logger.log project_id: project_id, newName:newName, "renaming project"
-		ProjectGetter.getProject project_id, {name:true}, (err, project)->
-			if err? or !project?
-				logger.err err:err,  project_id:project_id, "error getting project or could not find it todo project rename"
-				return callback(err)
-			oldProjectName = project.name
-			Project.update _id:project_id, {name: newName}, (err, project)=>
-				if err?
+		ProjectDetailsHandler.validateProjectName newName, (error) ->
+			return callback(error) if error?
+			logger.log project_id: project_id, newName:newName, "renaming project"
+			ProjectGetter.getProject project_id, {name:true}, (err, project)->
+				if err? or !project?
+					logger.err err:err,  project_id:project_id, "error getting project or could not find it todo project rename"
 					return callback(err)
-				tpdsUpdateSender.moveEntity {project_id:project_id, project_name:oldProjectName, newProjectName:newName}, callback
+				oldProjectName = project.name
+				Project.update _id:project_id, {name: newName}, (err, project)=>
+					if err?
+						return callback(err)
+					tpdsUpdateSender.moveEntity {project_id:project_id, project_name:oldProjectName, newProjectName:newName}, callback
+
+	MAX_PROJECT_NAME_LENGTH: 150
+	validateProjectName: (name, callback = (error) ->) ->
```

Comment:
if a name of "     " got through here without a trim it would pass the test. Worth doing a saftey trim and passing the trimmed name in the callback?